test/endpoints/favourite_filters/test_save_filter.py

```diff
@@ -0,0 +1,32 @@
+from bson import ObjectId
+from fastapi.testclient import TestClient
+
+from test.endpoints.favourite_filters.conftest import FavouriteFilter
+
+
+class TestSaveFilter:
+    def test_save_filter(
+        self,
+        test_app: TestClient,
+        login_and_get_token,
+    ):
+        name = "Test Create Filter"
+        filter_content = "PM-201-HJ-CRY-FLOW > 10"
+
+        test_response = test_app.post(
+            f"/users/filters?name={name}&filter={filter_content}",
+            headers={"Authorization": f"Bearer {login_and_get_token}"},
+        )
+
+        assert test_response.status_code == 201
+
+        # Testing the response is a valid object ID
+        filter_id = test_response.json()
+        ObjectId(filter_id)
+
+        test_filter = FavouriteFilter()
+        result = test_filter.find_one(filter_id)
+        print(f"Result 123: {result}")
+        assert result.get("filters") is not None
+
+        test_filter.delete(filter_id)
```

Comment:
If the test fails on an assertion, this line won't be reached. Either this should be in a fixture, or we should wrap the test in `try` and put the deletion in a `finally` block to ensure it's always run.

Reply:
Good spot - a fixture isn't appropriate as you can't pass data to them and you don't know what `filter_id` will be until the test executes. I've used the `finally` syntax, the change is in a03399c144cca1fd96d0954cb5ec276f5b8aca18